src/card.rs

```diff
@@ -0,0 +1,260 @@
+
+#[cfg(feature = "serialize")]
+use arrayvec::ArrayVec;
+#[cfg(feature = "serialize")]
+use bincode;
+#[cfg(feature = "serialize")]
+use serde::{Deserialize, Serialize};
+use core::i32;
+use core::result;
+
+const MIN_SAVE_AMNT: usize = 0x2000;
+const FILE_NAME_MAX: usize = 32;
+const CARD_ICON_MAX: usize = 8;
+#[allow(dead_code)]
+const CARD_ICON_WIDTH: usize = 32;
+#[allow(dead_code)]
+const CARD_ICON_HEIGHT: usize = 32;
+#[allow(dead_code)]
+const CARD_BANNER_WIDTH: usize = 96;
+#[allow(dead_code)]
+const CARD_BANNER_HEIGHT: usize = 32;
+
+macro_rules! card_try {
+    ($e:expr, $id:expr) => {
+        let result: CardError = unsafe { $e.into() };
+        if result == CardError::Ready {
+            report!("Card {}: {:#?}", $id, result);
+        } else {
+            report!("Card {} failed. -> {:?}", $id, result);
```

Comment:
I'd prefer to either not do any reports or gate them behind conditional compilation, like using debug_assert! or maybe a feature for this.

Reply:
The reports were really just there for me I guess. I've removed them